src/SDK/SmartDetectors.Extensions/AlertExtensions.cs

```diff
@@ -245,26 +262,26 @@ private static IEnumerable<AlertProperty> CreateAlertProperties(object propertyO
                     break;
 
                 case KeyValuePropertyAttribute keyValueAttribute:
-                    if (!(propertyValue is IDictionary<string, string> keyValuePropertyValue))
-                    {
-                        throw new ArgumentException($"A {nameof(KeyValuePropertyAttribute)} can only be applied to properties of type IDictionary<string, string>");
-                    }
+                    yield return CreateKeyValueAlertProperty(propertyOwner, propertyValue, propertyName, displayName, keyValueAttribute, order);
+                    break;
+
+                case TablePropertyAttribute tableAttribute:
+                    yield return CreateTableAlertProperty(propertyValue, propertyName, displayName, tableAttribute, order);
+                    break;
 
-                    if (keyValueAttribute.ShowHeaders)
+                case AzureResourceManagerRequestPropertyAttribute armRequestAttribute:
+                    if (!(propertyValue is AzureResourceManagerRequest armRequest))
                     {
-                        string keyHeaderName = keyValueAttribute.KeyHeaderName.EvaluateInterpolatedString(propertyOwner);
-                        string valueHeaderName = keyValueAttribute.ValueHeaderName.EvaluateInterpolatedString(propertyOwner);
-                        yield return new KeyValueAlertProperty(propertyName, displayName, order.Next(), keyHeaderName, valueHeaderName, keyValuePropertyValue);
+                        throw new ArgumentException($"A {nameof(AzureResourceManagerRequestPropertyAttribute)} can only be applied to properties of type {nameof(AzureResourceManagerRequest)}");
                     }
-                    else
+
+                    List<AlertProperty> propertiesToDisplay = armRequest.ExtractProperties().Where(prop => prop.PropertyName != nameof(AzureResourceManagerRequest.RequestUri)).ToList();
```

Comment:
>prop => prop.PropertyName != nameof(AzureResourceManagerRequest.RequestUri) [](start = 99, length = 75)

Nice! However, it will break if we add more properties to AzureResourceManagerRequest.
Consider passing an optional filter to ExtractProperties, and filter out all properties of AzureResourceManagerRequest.

Reply:
goot catch

---
In reply to: [279191205](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/61#discussion_r279191205) [](ancestors = 279191205)